### PR TITLE
Improve the developer experience.

### PR DESCRIPTION
**Pull request by markusthoemmes**
- Clearly spell out the requirements for the scripts.
- Remove minikube docs to not confuse people.
- Move `envsubst` call into the catalog.sh script so it produces a meaningful result as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,52 +1,62 @@
...
+`CatalogSource` (given you have a setup as described in the development section
+above) comprised of the `ClusterServiceVersions`, `CustomResourceDefinitions`,
+and package manifest in the bundle beneath [olm-catalog/](olm-catalog/). You
+should apply its output in the namespace where the other `CatalogSources` live
+on your cluster, e.g. `openshift-marketplace`:
 
 ```
-CS_NS=$(kubectl get catalogsources --all-namespaces | tail -1 | awk '{print $1}')
-./hack/catalog.sh | kubectl apply -n $CS_NS -f -
+CS_NS=$(oc get catalogsources --all-namespaces | tail -1 | awk '{print $1}')
```
Comment (nak3) [inline README.md]: Can we explicitly use `openshift-marketplace` or `openshift-operators` namespace? `tail -1` is fragile.

Comment (markusthoemmes) [reply README.md]: I'd kinda like to keep it the way it is as it's been there before. I want to ship some more pressing improvements first.

Comment (nak3) [reply README.md]: I see. No problem. then.

File: README.md [hunk 2]
```diff
@@ -1,52 +1,69 @@
...
 pushed to your docker repository.
 
-After that is done, all the scripts in the `hack` directory are at your disposal
-to install and test the system. `make test-e2e` in particular runs the entirety
-of the end-to-end tests against the system.
+### Installing the system/running tests
+
+Use the appropriate make targets or scripts in `hack`:
+
+- `make dev`: Deploys the servleress-operator without deploying Knative Serving.
```
Comment (mgencur) [inline README.md]: Typo: `servleress-operator`

File: README.md [hunk 3]
```diff
@@ -1,52 +1,69 @@
...
+and package manifest in the bundle beneath [olm-catalog/](olm-catalog/). You
+should apply its output in the namespace where the other `CatalogSources` live
+on your cluster, e.g. `openshift-marketplace`:
 
 ```
-CS_NS=$(kubectl get catalogsources --all-namespaces | tail -1 | awk '{print $1}')
-./hack/catalog.sh | kubectl apply -n $CS_NS -f -
+CS_NS=$(oc get catalogsources --all-namespaces | tail -1 | awk '{print $1}')
+./hack/catalog.sh | oc apply -n $CS_NS -f -
 ```
```
Comment (mgencur) [inline README.md]: Are these two commands above the same as `make dev` ? Is it worth mentioning here?

Comment (markusthoemmes) [reply README.md]: I'll keep these as-is for now as "detailed" steps how this interacts with OLM.

Comment (mgencur) [reply README.md]: Ah, `make dev` also installs service mesh. Maybe we could have a Makefile target called `make catalog` ? and then have the `make dev` call this inside.

Comment (markusthoemmes) [reply README.md]: Maybe, I'd like to keep that for further iterations in vain of getting the improvements made in this PR into the repo to unblock a few people.

File: README.md (hunk 1 above)
Comment (aliok) [reply README.md]: I think if you want to target OLM installed on Minikube, the namespace is called `olm`.
Glad you didn't change this part of the script

**Pull request merged**